components/prism-abap.js

```diff
@@ -0,0 +1,50 @@
+Prism.languages.abap= {
+	'comment': {
+		pattern: /^[\*].*$/gm,
+		lookbehind: false
+	},
+	'string' : /(`|')(\\?.)*?\1/gm,
+	'string_template': {
+	  pattern: /(\||\})(\\?.)*?(?=\||\{)/g,
+	  lookbehind: true,
+	  alias: 'string'
+	},
+	/* End Of Line comments should not interfere with strings when the  
+	quote character occurs within them. We assume a string being highlighted
+	inside an EOL comment is more acceptable than the oposite.
+	*/
+	'eol_comment': {
+		pattern: /(\s\").*$/gm,
+		lookbehind: false,
```

Comment:
Also, I think the `$` is useless, like I said before, because the `.*` pattern will consume characters until the end of line.